ddl/reorg.go

```diff
@@ -206,25 +196,23 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 			Location:      &model.TimeZoneLocation{Name: time.UTC.String(), Offset: 0},
 		}
 	}
-	if w.reorgCtx.doneCh == nil {
+	rc := w.getReorgCtx(job)
+	if rc == nil {
 		// Since reorg job will be interrupted for polling the cancel action outside. we don't need to wait for 2.5s
 		// for the later entrances.
 		// lease = 0 means it's in an integration test. In this case we don't delay so the test won't run too slowly.
 		if lease > 0 {
 			delayForAsyncCommit()
 		}
-		// start a reorganization job
+		rc = w.newReorgCtx(reorgInfo)
+		// this job is cancelling, we should notify the reorg worker.
+		if job.IsCancelling() {
```

Comment:
When will we encounter this logic？It(`rc == nil`) seems that we prepare start running backfill job which means no backfill workers need to cancel.

Reply:
yes, please see `rollingbackAddIndex`.

consider we are trying to cancel an add index job, and at the same time the job is going to the last round, the cancel job txn commit first and run ddl job txn will get a write conflict and retry. The reorg context is remove by the last round so at the retry round it will has `rc==nil` and job canceling

Reply:
it seems we can put this `if` before `rc==nil` stmt🤔